### PR TITLE
ldid 2.1.5-procursus7

### PR DESCRIPTION
**Pull request by TheRealKeto**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

For a while now, the original creator of ldid ("saurik") has left the project largely unmaintained in several areas. The intention of this change is to introduce a maintained version that's modern, while keeping up with changes coming from the original creator once in a while.

In particular, this new version contains changes that have improved software within the jailbreaking community (such as [Theos](https://theos.dev/)), being adopted as the version to use on modern versions of iOS, macOS, and Linux.

### CODE REVIEW

**Comment by carlocab** [comment]
Is the fork official or otherwise "blessed" by upstream?

**Comment by p-linnane** [comment]
According to [Repology](https://repology.org/project/ldid/versions) AUR is using this upstream, but MacPorts and nix are not. I do not support this change unless we get confirmation from @saurik that this is blessed as the new upstream.

**Comment by TheRealKeto** [comment]
Thanks for the review!

I'd like to make clear that I speak for myself, and am not representing the views of the Procursus Team or any organization, party, or group.

It's highly unlikely that @saurik will bless this change.

I initially believed in replacing the current version of ldid with the proposed change primarily because saurik has, through several means, [expressed disinterest in maintaining **some** of his software that the jailbreaking community depends on](https://twitter.com/saurik/status/1119211953866457089?s=21) (going as far as to deprecating Cydia entirely). This left a gap in several of his works (particularly ldid), with many forks and iterations springing up.

To centralize and keep his work afloat (such as Telesphoreo and ldid), much of the work the Procursus Team (and its contributors) have done has began to be regarded as "mainstream", as its main goal is to support further versions of iOS, where this utility shines the most.

While I believe in replacing the utility with this new upstream, I've failed to understand the concern for keeping an older, widely outdated version. Software, such as [Theos](https://theos.dev/), have began to adopt this modern version, and with it supporting Homebrew as one of many ways to install its dependencies, it would be convenient for Homebrew (and similar projects) to adopt this change. Know that this isn't me disrespecting any decision made, I would just like more insight should you feel compelled to share.

As a final thought, would proposing a new formula for this specific version of the project be allowed to be on Homebrew?

**Comment by p-linnane** [comment]
> It's highly unlikely that @saurik will bless this change.

Without that blessing we won't be able to change the upstream in this formula.

> While I believe in replacing the utility with this new upstream, I've failed to understand the concern for keeping an older, widely outdated version. Software, such as [Theos](https://theos.dev/), have began to adopt this modern version, and with it supporting Homebrew as one of many ways to install its dependencies, it would be convenient for Homebrew (and similar projects) to adopt this change. Know that this isn't me disrespecting any decision made, I would just like more insight should you feel compelled to share.

It is not about keeping an older version. It's about keeping the "official" version. Given the number of users and visibility Homebrew has, we are very strict about changing upstreams. We usually don't do so unless the original upstream blesses a new upstream, or other packagers have changed to a new upstream. Here's a case where I changed an upstream recently, with the reasoning in the PR: https://github.com/Homebrew/homebrew-core/pull/135879.

> As a final thought, would proposing a new formula for this specific version of the project be allowed to be on Homebrew?

Much like my previous point, if a fork is accepted by other packagers we will consider it.

**Comment by TheRealKeto** [comment]
> Much like my previous point, if a fork is accepted by other packagers we will consider it.

Is the AUR enough to justify adding a new formula with this specific version? Ideally, this version can be made available to other platforms, but should I do it myself would require more time before this PR can be merged.

I'd very much like this to be a done deal so that the process of adding it to other packagers is not halted.

**Comment by carlocab** [comment]
I'm fine with adding this as a different formula. If the name of the new formula could be endorsed by @ProCursusTeam that'd be helpful to avoid a situation where we have multiple different packages of their fork (packaged by different package managers) with different names.

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]
As above but to be explicit with a review:
- only willing to change this is Saurik gives explicit 👍🏻 
- would need a new homepage
- would need a new, stable tag

Alternatively:
- could add as a new formula e.g. `procursus-ldid` or `ldid-procursus` or similar
- could deprecate this formula if it meets our usual unmaintained criteria
- could mention new formula in the caveats for this formula

**Comment by CRKatri** [comment]
> * could add as a new formula e.g. `procursus-ldid` or `ldid-procursus` or similar

Adding a new `ldid-procursus` seems like the best option to me.

> * could deprecate this formula if it meets our usual unmaintained criteria

In my opinion, Saurik's main version of ldid is not unmaintained. He has still updated it in the past, this fork was created because Saurik does not accept patches, not due to it being abandoned.

**Comment by MikeMcQuaid** [comment]
> In my opinion, Saurik's main version of ldid is not unmaintained. He has still updated it in the past, this fork was created because Saurik does not accept patches, not due to it being abandoned.

Yes, let's definitely not deprecate or merge this then, thanks!

**Comment by p-linnane** [comment]
Closing this out based off maintainer feedback. Please open a new PR for `ldid-procursus` and we are happy to take a look.

**Pull request closed without merging**
